### ISSUE TITLE
ProxySQL Failed set to shunned reader when standalone 

### ISSUE THREAD

**Issue by lenovore** [signal18/replication-manager#455]
Replication Manager : v2.2.29
ProxySQL: v2.4.4

1. Can failover pre script and failover post script add another parameter to distinguish the caller (switchover or failover)
2. ProxySQL have 2 questions:
```
db-servers-hosts = "192.168.56.101,192.168.56.102"
db-servers-credential = "replmanager:replmanager"
db-servers-connect-timeout = 5

proxysql = true
proxysql-servers = "192.168.56.201,192.168.56.202"
proxysql-port = "6000"
proxysql-admin-port = "6032"
proxysql-writer-hostgroup = "1000"
proxysql-reader-hostgroup = "1001"
proxysql-user = "remoteadmin"
proxysql-password = "remoteadmin"
proxysql-bootstrap = false
proxysql-bootstrap-hostgroups = false
proxysql-bootstrap-users = false
proxysql-bootstrap-variables = false
proxysql-bootstrap-query-rules = false
proxysql-multiplexing = false
```
- failover error
![repl2](https://user-images.githubusercontent.com/29646774/203715160-c5555d22-195a-492c-b7a5-1eabd3536898.jpg)
- rejoin failed causes the old master to break away from the replication topology but proxysql old master status turn to ONLINE
![repl3](https://user-images.githubusercontent.com/29646774/203716563-4f28ed94-1bbe-4db0-96da-820af01e107a.jpg)

![proxysql4](https://user-images.githubusercontent.com/29646774/203714898-1e3776ac-fb01-4f2f-b9aa-fdd6875c7f98.jpg)

so, whether failover should set the old master in proxysql to offline_hard?
`update mysql_ servers set status='offline_ hard' where hostname='old matser'`

**Comment by svaroqui** [comment]
Thanks for reporting,
i'll try to reproduce and fixe it for failover pre script and failover post script adding a extra looks param like a good idea i'll work on it as well

**Comment by svaroqui** [comment]
Can you let me know what MySQL release you are using please ?
And what type of replication regular or GTID , our last dev branch support group.replication if you are interested in this feature

**Comment by svaroqui** [comment]
Not able to reproduce so far 
![Capture d’écran 2022-11-24 à 09 25 35](https://user-images.githubusercontent.com/2969888/203731568-fb457273-e5df-41be-981a-3081c79249f7.png)
![Capture d’écran 2022-11-24 à 09 29 41](https://user-images.githubusercontent.com/2969888/203731743-4b43fb89-2d48-40df-b70b-18637c195cdd.png)

Is the old master read only after rejoin ?

**Comment by svaroqui** [comment]
I see ou are using custom host group id , i do not in my regression test , let me investigate more

**Comment by svaroqui** [comment]
Can you let me know what is in  select * from mysql_replication_hostgroups; ?

**Comment by svaroqui** [comment]
Ok i  found one extra issue with proxysql when  bootstraping users , i don't pass the custom write group to user creation 
 
```
SELECT username, default_hostgroup FROM mysql_users;
+------------------+-------------------+
| username         | default_hostgroup |
+------------------+-------------------+
| mysql.session    | 0                 |
| mysql.sys        | 0                 |
| root             | 0                 |
| mysql.infoschema | 0                 |
+------------------+-------------------+

```
Please fixe by hand those tables until i produce a patch 
```

update mysql_replication_hostgroups set writer_hostgroup=1000 , reader_hostgroup=1001;
update mysql_users set  default_hostgroup=1000;
load mysql users to runtime;
```

**Comment by lenovore** [comment]
MySQL version is 5.7.40 and the replication topology is master slave semi-synchronized using GTID.

Yes, I use custom read-write group.
![image](https://user-images.githubusercontent.com/29646774/203914556-1c3899dd-a845-4738-b9bd-a1f0908bb866.png)

**Comment by svaroqui** [comment]
When you go to the proxy tab on web GUI do you see the state of the backends ?

**Comment by lenovore** [comment]
after failover
![proxysql2](https://user-images.githubusercontent.com/29646774/203996660-8f35ae8f-c34d-4174-b3f3-3da26115c89e.jpg)
after rejoin failed
![repl3](https://user-images.githubusercontent.com/29646774/203996793-2e0430c1-2749-4cb2-aa41-5a23902b56c0.jpg)

![proxysql3](https://user-images.githubusercontent.com/29646774/203996748-b0e22fce-7c05-47b2-a81e-5dd2dc11b75d.jpg)

**Comment by lenovore** [comment]
My proxysql issue
https://github.com/sysown/proxysql/issues/4025

**Comment by lenovore** [comment]
I use `reset slave all` to represent `rejoin failed`

**Comment by lenovore** [comment]
When old master rejoin failed,replication manager is standalone,but proxysql is online.

**Comment by svaroqui** [comment]
Ok i may have found something and will build a new release 2.2.30 containing also the additional parameter for failover switchover script  but can you let me know if you change the backend state to shunned , does proxysql reset it to online ?

**Comment by svaroqui** [comment]
I succeed to reproduce the issue and indeed this state was no more checked on reader group but only on writer group, This  regression was introduced during implementation of dynamic reload config and writer is also reader and was not happening during the failover itself but after replication-manager is restarted i was able to reproduce same issue  and it should be fixed in 2.2.31 please reopen if not fixed . FYI there is an  option proxysql-debug=true to follow what is triggered by replication-manager regarding proxysql state tracking.  Thanks  a lot
